di2f/index.md

```diff
@@ -0,0 +1,226 @@
+
+In addition to exposing more "traditional" web2 APIs (e.g., REST, gRPC, etc.) as services, it is possible to expose these APIs over p2p protocols. For example, it is possible to allow peers to serve HTTP endpoints and make HTTP requests through `libp2p` using Go's standard "http" and "net" stack. This provides a simple on-ramp for web 2.0 developers to expose their APIs over the threads p2p network. Couple service provision and remote database access with the robust authentication and globally identifiable assets afforded by thread DIDs, and we now have a very clear path to onboarding web2 developers (and data) to the decentralized web.
+
+# Conclusions
+
+Decentralized identifiers (DIDs) enable cross-protocol/blockchain interactions by defining a common interface to retrieve and validate entities, such as users and data. A DID-driven threads network offers an exciting opportunity to expose p2p services as first-class components of a p2p system.
+
+Threads and IPFS/Filecoin benefit from a native DID in numerous ways. Firstly, DIDs provide globally unique namespaces for resources, such that thread IDs act as identifiers to network-wide resources. Secondly, adopting a standardized system for resource identification increases ecosystem(s) interoperability, facilitating integration with external DID methods available on systems such as Filecoin or Ethereum, as well as a range of identity solutions (e.g., Ceramic/IDX). Thirdly, globally unique identifiers pave the way for _distributed authorization_ to network resources, because access control mechanisms are globally and unambiguously defined. This leads to a net increase in the decentralization of network services.
+
+An additional key benefit of the proposed system includes the provision of "native" discovery mechanisms. Network peers can do work on each others' behalf since any peer is capable of discovering and resolving thread DID namespaces. Additionally, since all operations on a thread lead to deterministic, content-addressable updates, any peer on the network can participate in the update of a thread to which they have access.
+
+<!-- @sanderpick this is pretty hand-wavy here, needs some help! -->
+
+There are additional motivators for exposing a DID-friendly threads specification. One such motivator is providing "crypto-native" access to off-chain data. Decentralized app developers can enable their users to directly leverage local crypto wallets to interact with thread-enabled apps. This includes Filecoin integration. Threads then become append-only logs that can be anchored to Filecoin manually or at some frequency. Buckets (derivatives of threads) can expose a mutable filesystem and pinning API that can be similarly archived to Filecoin. In this sense, DID-based threads can become a Filecoin "Layer 2" protocol.
+
+<!-- @sanderpick This last paragraph is similarly hand-wavy, and maybe not the topic we want to end on? -->
```

Comment:
@sanderpick This last paragraph is similarly hand-wavy, and maybe not the topic we want to end on?

Reply:
This could be swapped for something about service discovery mentioned above.